File: CPP/Inheritance.java
//									PROGRAM TO DEMONSTRATE INHERITANCE (SINGLE INHERITANCE)

class A					//Base Class
{
		int a;
	
		A()
		{
			a=10;
		}
};

class B extends A			//Derived Class (Class B inherits the public properties of Class A)
{
		int b;
	
		B()
		{
			b=20;
		}
		void displayAB()
		{
			System.out.println("a=" + a + ", b=" + b);
		}	
};

public class Inheritance
{ 
	public static void main(String[] args)
	{
		B obB = new B();
	
		obB.displayAB();
	}
}


/*
	In C++ object may be created statically (Compile-time) or dynamically (Runtime) 
	whereas in java it is created dynamically
	
*/

File: VB/Inheritance.vb
REM									PROGRAM TO DEMONSTRATE INHERITANCE (SINGLE INHERITANCE)

Class A					'Base Class

	Protected a as integer
		
	Public Sub New()		'Constructor		
		a=10
	End Sub
End Class

Class B 			'Derived Class (Class B inherits the public properties of Class A)
	Inherits A
	
	Private b as integer
	
	public Sub New()		'Constructor
		b=20
	End Sub
	Public Sub DisplayAB()
		Console.WriteLine("a=" & a & ", b=" & b)
	End Sub
End Class

Module Inheritance
	Sub Main()
		Dim obB as new B()
	
		obB.DisplayAB()
	End Sub
End Module
